dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -297,6 +297,58 @@ HRESULT CDrivesContextMenu_CreateInstance(PCIDLIST_ABSOLUTE pidlFolder,
     return CDefFolderMenu_Create2(pidlFolder, hwnd, cidl, apidl, psf, DrivesContextMenuCallback, cKeys, hKeys, ppcm);
 }
 
+static BOOL
+getAutoRunInfo(LPCWSTR Entry, LPWSTR pszValue, DWORD cchValueLen, LPCWSTR InfFile)
+{
+    static const WCHAR s_icon[] = { 'i', 'c', 'o', 'n', 0 };
+    return GetPrivateProfileStringW(Entry, s_icon, NULL, pszValue, cchValueLen, InfFile);
+}
+
+static HRESULT
+getIconLocationForDrive(IShellFolder *psf, PCITEMID_CHILD pidl, UINT uFlags,
+                        LPWSTR szIconFile, UINT cchMax, int *piIndex, UINT *pwFlags)
+{
+    WCHAR wszPath[MAX_PATH];
+    WCHAR wszAutoRunInfPath[MAX_PATH];
+    WCHAR wszValue[MAX_PATH], wszTemp[MAX_PATH];
+    static const WCHAR wszAutoRunInf[] = { 'a','u','t','o','r','u','n','.','i','n','f',0 };
+    static const WCHAR wszAutoRun[] = { 'a','u','t','o','r','u','n',0 };
+
+    // get path
+    if (!ILGetDisplayNameExW(psf, pidl, wszPath, 0))
+        goto Quit;
```

Comment:
There is not really something to clean up, so `return E_FAIL;` would be preferred here (imo)